node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/text/SnippetGenerator.java

```diff
@@ -0,0 +1,30 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.node.admin.task.util.text;
+
+/**
+ * @author hakon
+ */
+public class SnippetGenerator {
```

Comment:
I spent way too much time trying to optimize the original version on Friday, but couldn't get rid of the `while` loop while keeping the result perfect. But you can get it within 1 character error by doing `int assumedDigits = (int) (Math.log10(text.length() - sizeHint) + 1);` and then remove the `while` loop.